packages/common/util.js

```diff
@@ -31,11 +31,16 @@ const log = require('./log');
  *
  * @alias module:util
  */
+const warned = new Set();
 exports.deprecate = (name, version, alternative) => {
-  let message = `${name} is deprecated after version ${version} and will be removed in a future release.`;
-  if (alternative) message += ` Use ${alternative} instead.`;
-  if (!('NO_DEPRECATION_WARNINGS' in process.env)) {
-    log.warn(message);
+  const key = `${name}-${version}`;
+  if (!warned.has(key)) {
+    warned.add(key);
+    let message = `${name} is deprecated after version ${version} and will be removed in a future release.`;
+    if (alternative) message += ` Use ${alternative} instead.`;
+    if (process.env.NO_DEPRECATION_WARNINGS !== 'true') {
+      log.warn(message);
+    }
```

Comment:
Thanks for this. This will massively reduce output even when warnings are not disabled.

This should also be updated in `packages/aws-client/src/StepFunctions.ts`, where `deprecate` is duplicated. I don't know if there are other places.